migrate badge documentation from pull into wiki (#83)
louislam/uptime-kuma-wiki: Badge.md

- `prefix`
- `suffix`, Default value: uptime= `%`, ping, avg-response & response= `ms`
- `color`
- `labelColor`

**Examples**

| Badge | Url |
|-------|-----|
| ![image](img/badge/ping-label.png) | `<kuma-url>/api/badge/<monitorId>/ping/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>` |
| ![image](img/badge/uptime-label.png) | `<kuma-url>/api/badge/<monitorId>/uptime/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>` |
|  | `<kuma-url>/api/badge/<monitorId>/avg-response/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>` |

The default uptime badge's color is based on the uptime percentage:

![image](img/badge/uptime-color.png)

but can be overwritten with [predefined color](https://www.npmjs.com/package/badge-maker#colors) or any other color in HEX code:

![image](img/badge/custom-colours.png)

### Badge Duration at ping, uptime and avg-response

The given time interval at `Badge Duration (in hours)` has to be in 'hours', but customizing options allow for a different display: 

> **Example***

Reviewer: Noticed there was a typo added by me ^^

```suggestion
**Example**
```

(Version >= 1.16.0)

> With https://github.com/louislam/uptime-kuma/pull/1119

Reviewer: Other sections of the wiki also don't have the PR they are added in

```suggestion
```

Author: ok, so we delete here

> - [Description](#description)
> - [Open Badge Generator](#open-badge-generator)
> - [Badge Type Options](#badge-type-options)
>   - [Status badge](#status-badge)
>   - [Ping, uptime, avg-response and response badge](#ping-uptime-avg-response-and-response-badge)
>     - [Badge Duration at ping, uptime and avg-response](#badge-duration-at-ping-uptime-and-avg-response)
>   - [Certificate Expire badge](#certificate-expire-badge)
> - [Badge Styles](#badge-styles)

Reviewer: Rationale: other pages of the wiki also don't have such a section

```suggestion
```

Author: you are right, a toc is no longer needed at github.com, there is the "Outline" feature, which creates the same thing.

[https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/](https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/)

> - [Badge Styles](#badge-styles)

Reviewer: I think this can be pruned as the other wiki pages also don't have this

```suggestion
```

|----------------|---------------------------------------------------------|---------------------------------------|----------------------------------------------|
| `status`       | `<kuma-url>/api/badge/:monitorID/status`                | `<kuma-url>/api/badge/1/status`       | ![image](img/badge/status-example.png)       |
| `uptime`       | `<kuma-url>/api/badge/:monitorID/uptime:duration`       | `<kuma-url>/api/badge/1/uptime`       | ![image](img/badge/uptime-example.png)       |
| `ping`         | `<kuma-url>/api/badge/:monitorID/ping:duration`         | `<kuma-url>/api/badge/1/ping`         | ![image](img/badge/ping-example.png)         |
| `avg-response` | `<kuma-url>/api/badge/:monitorID/avg-response:duration` | `<kuma-url>/api/badge/1/avg-response` | ![image](img/badge/avg-response-example.png) |
| `cert-exp`     | `<kuma-url>/api/badge/:monitorID/cert-exp`              | `<kuma-url>/api/badge/1/cert-exp`     | ![image](img/badge/cert-exp-example.png)     |
| `response`     | `<kuma-url>/api/badge/:monitorID/response`              | `<kuma-url>/api/badge/1/response`     | ![image](img/badge/response-example.png)     |

> # Open Badge Generator
> 

Reviewer: Don't think this tip needs a title ^^

```suggestion
```

Author: ok, so we delete this headline

> [!TIP]
> We have an Badge Generator integrated which makes this configuration more interactive.
> 
> You can use said generator via: 
> - navigating to a status page in edit mode
> - <details><summary>Clicking on the settings icon</summary>
>   <p>
>   
>   ![image](img/badge/settings-button.png)
>   
>   </p>
>   </details>
> - <details><summary>Clicking on the <code>Open Badge Maker</code>-button</summary>
>   <p>
>   
>   ![image](img/badge/settings.png)
>   
>   </p>
>   </details>
> - <details><summary>filling out the form to create your own badge</summary>
>   <p>
>   
>   ![image](img/badge/open-badge-generator.png)
>   
>   </p>
>   </details>

# Badge Type Options

## Status badge

There are options to customize the **status** badge's appearance: 
- `upLabel`,Default value: Up
- `downLabel`, Default value: Down
- `upColor` 
- `downColor`

> **Example**
> 
> `<kuma-url>/api/badge/<monitorId>/status?upLabel=<upLabel>&downLabel=<downLabel>` ![image](img/badge/status-label.png)
> 
> `<kuma-url>/api/badge/<monitorId>/status?upColor=white&downColor=black` ![image](img/badge/status-bw.png)

Reviewer: ```suggestion
**Examples**

| Badge | Url |
---------------
| ![image](img/badge/status-label.png) | `<kuma-url>/api/badge/<monitorId>/status?upLabel=<upLabel>&downLabel=<downLabel>` |
| ![image](img/badge/status-bw.png) | `<kuma-url>/api/badge/<monitorId>/status?upColor=white&downColor=black` |
```

## Ping, uptime, avg-response and response badge

There are options to customize the **ping**, **uptime**, **avg-response** and **response** badge's text:
- `labelPrefix`
- `label`
- `labelSuffix`, Default: value: uptime, ping & avg-response 	image= `h`
- `prefix`
- `suffix`, Default value: uptime= `%`, ping, avg-response & response= `ms`
- `color`
- `labelColor`

> **Example**
> 
> `<kuma-url>/api/badge/<monitorId>/ping/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>`
> 
> ![image](img/badge/ping-label.png)
> 
> `<kuma-url>/api/badge/<monitorId>/uptime/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>`
> 
> ![image](img/badge/uptime-label.png)
> 
> `<kuma-url>/api/badge/<monitorId>/avg-response/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>`

Reviewer: > [!NOTE]
> there is no example image for the third link

```suggestion
**Examples**

| Badge | Url |
---------------
| ![image](img/badge/ping-label.png) | `<kuma-url>/api/badge/<monitorId>/ping/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>` |
| ![image](img/badge/uptime-label.png) | `<kuma-url>/api/badge/<monitorId>/uptime/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>` |
|  | `<kuma-url>/api/badge/<monitorId>/avg-response/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>` |
```

The default uptime badge's color is based on the uptime percentage:

![image](img/badge/uptime-color.png)

> but can be overwritten with predefined color or any other color in HEX code:
> 
> ![image](img/badge/custom-colours.png)
> 
> See https://www.npmjs.com/package/badge-maker#colors for predefined colors.

Reviewer: ```suggestion
but can be overwritten with [predefined color](https://www.npmjs.com/package/badge-maker#colors) or any other color in HEX code:

![image](img/badge/custom-colours.png)
```

### Badge Duration at ping, uptime and avg-response

The given time interval at `Badge Duration (in hours)` has to be in 'hours', but customizing options allow for a different display: 

> `<kuma-url>/api/badge/1/uptime/720?label=30&labelSuffix=d`
> 
> ![image](img/badge/duration.png)

Reviewer: ```suggestion
**Example***

| Badge | Url |
---------------
| ![image](img/badge/duration.png) | `<kuma-url>/api/badge/1/uptime/720?label=30&labelSuffix=d` |
```

## Certificate Expire badge

There are the same options like **ping**, **uptime** and **avg-response** and additional to customize the **cert-exp** badge's text: 
- `upColor`
- `downColor` 
- `warnDays` 
- `downDays`

# Badge Styles

The different badge styles as defined by https://shields.io/#styles. (thx @throwabird / [comment](https://github.com/louislam/uptime-kuma/pull/1119#issuecomment-1004760533) )

- `flat` (default)
- `flat-square`
- `plastic`
- `for-the-badge`
- `social`

> **Example**
> 
> `<kuma-url>/api/badge/<monitorId>/status?style=flat-square`
> 
> ![image](img/badge/badge-style.png)

Reviewer: ```suggestion
**Example**

| Badge | Url |
---------------
| ![image](img/badge/badge-style.png) | `<kuma-url>/api/badge/<monitorId>/status?style=flat-square` |
```

(Version >= 1.16.0)

- [Description](#description)
- [Status badge](#status-badge)
- [Ping and uptime badge](#ping-and-uptime-badge)
- [Screenshots](#screenshots)

# Description

Status, ping and uptime badges, generated in node-js, no internet access required (based on [badge-maker](https://www.npmjs.com/package/badge-maker) aka. [shields.io](http://shields.io/))

> There are three API endpoints that return an SVG graphic for a given (public) monitor:

Reviewer: Please include all badges, not only the ones from the intial PR
![image](https://github.com/louislam/uptime-kuma-wiki/assets/26258709/2a7359ed-117e-4375-bea3-9437ffe212c7)

> **Status** `<kuma-url>/api/badge/:monitorID/status` e.g. `<kuma-url>/api/badge/1/status` returning sth like: ![image](img/badge/148052351-08a5713f-2bdf-4034-ab2a-29c7638cb25c.png)

Reviewer: I think this would be clearer if presented as a table (=> less visual clutter)

**Ping** `<kuma-url>/api/badge/:monitorID/ping/:duration` e.g. `<kuma-url>/api/badge/1/ping` returning sth like: ![image](img/badge/148052406-55b55a76-f61b-4a2a-a036-53b557eebc49.png)

**Uptime** `<kuma-url>/api/badge/:monitorID/uptime/:duration` e.g. `<kuma-url>/api/badge/1/uptime/24` returning sth like: ![image](img/badge/147946965-5a706c7f-1efe-4056-b06b-b149904f87b5.png)

> # Status badge

Reviewer: I think we can skip these explainations and instead show the badge-maker. Currently, there are a few badges which are not included in this explaination like cert-expiry

Said feature is hidden here:
![image](https://github.com/louislam/uptime-kuma-wiki/assets/26258709/ec3267ce-6a2f-4653-bf51-3553be119b5e)
![image](https://github.com/louislam/uptime-kuma-wiki/assets/26258709/5dd6e68a-81cb-4358-b4cc-8f7633bc12a1)
![image](https://github.com/louislam/uptime-kuma-wiki/assets/26258709/67a16e8e-bc49-44af-82c2-ef0da83e5955)

(Version >= 1.16.0)

- [Description](#description)
  - [Status badge](#status-badge)
  - [Ping, uptime and avg-response badge](#ping-uptime-and-avg-response-badge)
  - [Certificate Expire badge](#certificate-expire-badge)
  - [Response badge](#response-badge)
  - [Badge Styles](#badge-styles)
  - [Open Badge Generator](#open-badge-generator)
- [Screenshots](#screenshots)

# Description

Status, ping, uptime, avg-response badges, cert-exp and response generated in node-js, no internet access required (based on [badge-maker](https://www.npmjs.com/package/badge-maker) aka. [shields.io](http://shields.io/))

> There are six API endpoints that return an SVG graphic for a given (public) monitor:

Reviewer: Nit: Lets avoid adding a number that we need to update here ^^
(if we add another badge in the future and don't update this number, people will think we are morons that can't count to 7...)

```suggestion
These are the endpoints that return an SVG graphic for a given (public) monitor:
```

| Badge Type   | URL scheme                                            | URL example                         | Graphic example                              |
|--------------|-------------------------------------------------------|-------------------------------------|----------------------------------------------|
| status       | <kuma-url>/api/badge/:monitorID/status                | <kuma-url>/api/badge/1/status       | ![image](img/badge/status-example.png)       |
| uptime       | <kuma-url>/api/badge/:monitorID/uptime:duration       | <kuma-url>/api/badge/1/uptime       | ![image](img/badge/uptime-example.png)       |
| ping         | <kuma-url>/api/badge/:monitorID/ping:duration         | <kuma-url>/api/badge/1/ping         | ![image](img/badge/ping-example.png)         |
| avg-response | <kuma-url>/api/badge/:monitorID/avg-response:duration | <kuma-url>/api/badge/1/avg-response | ![image](img/badge/avg-response-example.png) |
| cert-exp     | <kuma-url>/api/badge/:monitorID/cert-exp              | <kuma-url>/api/badge/1/cert-exp     | ![image](img/badge/cert-exp-example.png)     |

> | response     | <kuma-url>/api/badge/:monitorID/response              | <kuma-url>/api/badge/1/response     | ![image](img/badge/response-example.png)     |

Author: ok, but why not at `Status` and `Uptime` at `URL-Schema` and `URL-Example`?

> | response     | <kuma-url>/api/badge/:monitorID/response              | <kuma-url>/api/badge/1/response     | ![image](img/badge/response-example.png)     |

Reviewer: What?

`<kuma-url>/api/badge/<monitorId>/avg-response/24?labelPrefix=<labelPrefix>&label=<label>&labelSuffix=<labelSuffix>&prefix=<prefix>&suffix=<suffix>`

The given time interval at `Badge Duration (in hours)` has to be in 'hours', but customizing options allow for a different display: `<kuma-url>/api/badge/1/uptime/720?label=30&labelSuffix=d`

![image](img/badge/duration.png)

The default uptime badge's color is based on the uptime percentage:

![image](img/badge/uptime-color.png)

but can be overwritten with predefined color or any other color in HEX code:

![image](img/badge/custom-colours.png)

See https://www.npmjs.com/package/badge-maker#colors for predefined colors.

Different badge styles as defined by https://shields.io/#styles are also supported. (thx @throwabird / [comment](https://github.com/louislam/uptime-kuma/pull/1119#issuecomment-1004760533) )

`<kuma-url>/api/badge/<monitorId>/status?style=flat-square`

## Certificate Expire badge

There are options to customize the **cert-exp** badge's text: `labelPrefix`, `label`, `labelSuffix`, `prefix`, `suffix = days`, `color`, `labelColor`, `upColor`, `downColor`, `warnDays`, `downDays`:

## Response badge

> There are options to customize the **response** badge's text: `labelPrefix`, `label`, `labelSuffix`, `prefix`, `suffix = ms / %`, `color`, `labelColor`:

Reviewer: I don't have any good ideas how to make this feel less like a (quite confusing) parameter dump.

I think these could work:
- explaining all the common options only once (`labelPrefix`, `label`, `labelSuffix`, `prefix`, `color`, `labelColor`). `Badge Styles` from below should be handled like this as well.
- Using a table for the remaining options

## Badge Styles

Use of the following options to enable them: `flat` (default), `flat-square`, `plastic`, `for-the-badge`, `social`: ![image](img/badge/badge-style.png)

> ## Open Badge Generator
> 
> Call up the Open Badge Generator of a service by pressing the Settings button on a status page in edit mode.
> 
> ![image](img/badge/settings-button.png)
> 
> ![image](img/badge/settings.png)
> 
> ![image](img/badge/open-badge-generator.png)

Reviewer: I think including this as a tip at the very top of the page would be better.
I have reworded some content to make this into a tip

Tips look like this:
![image](https://github.com/louislam/uptime-kuma-wiki/assets/26258709/3b2ac2bb-6b8a-41a4-8f56-14ff8c2c4f83)

```suggestion
> [!TIP]
> We have an Badge Generator integrated which makes this configuration more interactive.
> 
> You can use said generator via: 
> - navigating to a status page in edit mode
> - <details><summary>Clicking on the settings icon</summary>
>   <p>
>   
>   ![image](img/badge/settings-button.png)
>   
>   </p>
>   </details>
> - <details><summary>Clicking on the <code>Open Badge Maker</code>-button</summary>
>   <p>
>   
>   ![image](img/badge/settings.png)
>   
>   </p>
>   </details>
> - <details><summary>filling out the form to create your own badge</summary>
>   <p>
>   
>   ![image](img/badge/open-badge-generator.png)
>   
>   </p>
>   </details>
```

# Screenshots

![image](img/badge/examples.png)

> 

Author: we can delete it, there are enough pictures in the documentation
